src/Illuminate/Support/Collection.php

```diff
@@ -474,6 +476,40 @@ public function filter(callable $callback = null)
         return new static(array_filter($this->items));
     }
 
+    /**
+     * Apply the format in formatter given class.
+     *
+     * @param $class
+     * @return Collection
+     * @throws Exception
+     */
+    public function format($class)
+    {
+        $result = [];
+
+        foreach ($this->items as $key => $value) {
+            $formatter = new $class($value, $key);
+
+            if (! $formatter instanceof Formatter) {
+                throw new \Exception('The given class to format the data is not instance of Formatter');
```

Comment:
I would use the whole namespace here, Formatter is a pretty generic name.

Reply:
I agree, I think the exception should be more verbose